zeppelin-web-angular/src/app/pages/workspace/interpreter/interpreter.component.ts

```diff
@@ -46,7 +46,11 @@ export class InterpreterComponent implements OnInit, OnDestroy {
   }
 
   filterInterpreters(value: string) {
-    this.filteredInterpreterSettings = this.interpreterSettings.filter(e => e.name.search(value) !== -1);
+    const lowerCaseValue = value.toLowerCase();
+    this.filteredInterpreterSettings = this.interpreterSettings.filter(e =>
+      e.name.toLowerCase().includes(lowerCaseValue)
+    );
+
```

Comment:
```suggestion
    this.filteredInterpreterSettings = this.interpreterSettings.filter(e => e.name.search(new RegExp(value, 'i')) !== -1);

```
Consider using the 'i' flag with RegExp here. This way, I think we can support both Regex patterns and regular string searches while keeping it case-insensitive, which would closely match the original functionality.

Reply:
@tbonelee Makes sense. I've adapted it. [0727f10](https://github.com/apache/zeppelin/pull/4897/commits/0727f10d96060c83f6a68686ed3eba62821e4738)